lib/facter/util/resolvers/ffi/hostname.rb

```diff
@@ -36,19 +36,23 @@ def self.getffihostname
             raw_hostname.read_string
           end
 
-          def self.getffiaddrinfo(hostname)
-            hostname_ptr = FFI::MemoryPointer.new(hostname)
+          def self.getffiaddrinfo(hostname) # rubocop:disable  Metrics/AbcSize
+            ret = FFI::MemoryPointer.new(Facter::Util::Resolvers::Ffi::AddrInfo)
```

Comment:
```       int getaddrinfo(const char *node, const char *service,
                       const struct addrinfo *hints,
                       struct addrinfo **res);
```
based on this, ret, should be pointer to pointer, I assume that allocating a `:pointer` should be enough